kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -874,6 +874,31 @@ impl<'tcx> GotocCtx<'tcx> {
         Stmt::atomic_block(vec![skip_stmt], loc)
     }
 
+    // Computes the offset from a pointer
+    // https://doc.rust-lang.org/std/intrinsics/fn.offset.html
+    fn codegen_offset(
+        &mut self,
+        intrinsic: &str,
+        mut fargs: Vec<Expr>,
+        p: &Place<'tcx>,
+        loc: Location,
+    ) -> Stmt {
+        let src_ptr = fargs.remove(0);
+        let offset = fargs.remove(0);
+        // Check that the computation would not overflow an `isize`
+        let dst_ptr_of = src_ptr.clone().cast_to(Type::ssize_t()).add_overflow(offset.clone());
```

Comment:
This does integer arithmatic, not pointer arithmatic, and I think will give the wrong results

Reply:
I.e.: You should account for the pointee size.

Reply:
Right, should be getting the right results now.